stdlib/digest.mli

```diff
@@ -16,12 +16,12 @@
 (** MD5 message digest.
 
    This module provides functions to compute 128-bit 'digests' of
-   arbitrary-length strings or files. The digests are of cryptographic
-   quality: it is very hard, given a digest, to forge a string having
-   that digest. The algorithm used is MD5. This module should not be
-   used for secure and sensitive cryptographic applications. For these
-   kind of applications more recent and stronger cryptographic
-   primitives should be used instead.
+   arbitrary-length strings or files. The algorithm used is MD5.
+
+   The MD5 hash function is not cryptographically secure.
+   Hence, this module should not be used for security-sensitive
+   applications.  More recent, stronger cryptographic primitives
```

Comment:
A very minor nitpick: should we not recommend more recent libraries (with better cryptographic properties) rather than better cryptographic primitives?

Reply:
We have [mirage-crypto](https://github.com/mirage/mirage-crypto) which is maintained and has

Reply:
The sentence is about "the MD5 hash function", so it makes sense to talk of "better cryptographic primitives" -- or "better cryptographic hash functions" if you wish.

Shall we recommend alternate libraries that provide those better functions?  That's not something we usually do in the stdlib documentation, and there are many such libraries (do an OPAM search on "crypto"), so I wouldn't know which one(s) to mention.

Reply:
I read the paragraph with more emphasis on `this module`, and thus I was wondering about the back and forth `hash function -> this module -> cryptographic primitives` (with the supplementary issue that cryptographic primitives are generally not directly used by most users?). It is probably better indeed to not enshrine a specific external library.